src/mapping/RadialBasisFctMapping.hpp

```diff
@@ -145,8 +171,17 @@ void RadialBasisFctMapping<RADIAL_BASIS_FUNCTION_T>::computeMapping()
       globalOutMesh.addMesh(*outMesh);
     }
 
-    _rbfSolver = RadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T>{this->_basisFunction, globalInMesh, boost::irange<Eigen::Index>(0, globalInMesh.vertices().size()),
-                                                               globalOutMesh, boost::irange<Eigen::Index>(0, globalOutMesh.vertices().size()), this->_deadAxis, _polynomial};
+    if (_useEigen) {
+      _rbfSolver = RadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T>{this->_basisFunction, globalInMesh, boost::irange<Eigen::Index>(0, globalInMesh.vertices().size()),
+                                                                 globalOutMesh, boost::irange<Eigen::Index>(0, globalOutMesh.vertices().size()), this->_deadAxis, _polynomial};
+    }
+#ifndef PRECICE_NO_GINKGO
+    else {
+      _ginkgoRbfSolver.~GinkgoRadialBasisFctSolver();
+      new (&_ginkgoRbfSolver) GinkgoRadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T>{this->_basisFunction, globalInMesh, boost::irange<Eigen::Index>(0, globalInMesh.vertices().size()),
```

Comment:
Interesting syntax. I guess there is a good reason for that? We could make the whole object a `unique_ptr` in order to unify the syntax..

Reply:
This is very dangerous territory! Placement new and in-place destruction are meant for dealing with uninitialized memory.

We should consider (in this order) reconstruction, `.reset()`/`.clear()`, `std::optional` (if you don't like heap memory), or `std::unique_ptr`.

Reply:
So I finally checked again and my thoughts were that copying all the executor pointers etc. might be a bit unclean. I added a proper destructor deleting all pointers and calling `clear()` which frees the memory of all Ginkgo matrices now. Therefore, this "placement new" shouldn't make any problems anymore, or do you think we should still rather remove it?

Reply:
I would rather formulate the question the other way around: you are trying to use a rather inconvenient language feature: why should we do that? I cannot imagine a whole lot of performance benefits using this style (proof me wrong). We call this loc per simulation exactly once and the large matrices are only allocated once we we hand over the system-size information here.
 
Referring to the  discussion above (https://github.com/precice/precice/pull/1581#discussion_r1117195631), making the solver the actual template parameter will kind of merge the line here with the other `if` branch, so we anyway need to find a solution here working for 'all' solvers.

Reply:
I also think making it a template will solve this. Let me try to figure out a clean way to do this. We can then discuss this in person.